src/v/bytes/details/io_allocation_size.h

```diff
@@ -61,7 +61,8 @@ class io_allocation_size {
         if (size <= ss_max_small_allocation) {
```

Comment:
Can you update the comment with the detail that the result size is now capped by max_chunk?

It is actually quite confusing that we have `next_allocation_size` which does really give you the next (bigger) allocation size, and then this `ss_next_allocation_size` which does kind of the opposite: giving you the _smaller_ size (or equal for small sizes or exact power of two), but I guess that's not worth changing now (what do you think @dotnwat?).

Reply:
Names are indeed confusing.  I've embelished the comments for both, to make clear that one is for adding data when we know the total final size (ss_next_allocation_size) and the other is for adding data which may be followed by unknown quanitites of further data (next_allocation_size).

Reply:
Ah, I get it now that "next" here means "for the next chunk" (e.g., while copying a bunch of chunks in `copy`), not "next larger allocation size", i.e., "round up to an allocation boundary" as I had read it earlier (and why I was complaining about the names). I still maintain my complaint a little bit because the function should say what it does rather that being named for how the (primary) caller happens to use but but now I'm really into bikeshed territory.

Thanks for adding these comments and LGTM.